designs/2021-init-command-eslint-cli/README.md

```diff
@@ -0,0 +1,153 @@
+- Repo: [eslint/eslint](https://github.com/eslint/eslint)
+- Start Date: 2020-06-29
+- RFC PR: https://github.com/eslint/rfcs/pull/79
+- Authors: Aniketh Saha ([anikethsaha](https://github.com/anikethsaha)) aladdin-add(weiran.zsd@outlook.com)
+
+# Move --init flag into a separate utility
+
+## Summary
+
+- Remove the auto-config.
+- Move the `init` command from main repo ([eslint](https://github.com/eslint/eslint)) to a new repo named `@eslint/create-config`.
+
+## Motivation
+
+Currently the whole `init` command is being shipped with the cli in the main repo. Though its not that much of size (roughly ~0.6MB for the [eslint/lib/init](https://github.com/eslint/eslint/tree/master/lib/init)), this command is not a type of command that we need everyday or everytime running eslint. It is mainly used when creating a new project or adding eslint to a project for the first time. So if we move this to a separate package that is meant to use in the command line,
+We will make use of tools like `npm` to simply run `npm init @eslint/config` or using `npx` to run `npx  @eslint/create-config` or using `yarn` to run `yarn create @eslint/config` single time for a project instead of having it in the core project.
+
+## Detailed Design
+
+<!--
+   This is the bulk of the RFC.
+
+   Explain the design with enough detail that someone familiar with ESLint
+   can implement it by reading this document. Please get into specifics
+   of your approach, corner cases, and examples of how the change will be
+   used. Be sure to define any new terms in this section.
+-->
+
+The implementation is mainly migrating the code from main repo to `@eslint/create-config`.
+
+The approach would be following these steps
+
+### 1. Remove eslint auto-config
+
+- Remove auto-config(`eslint/lib/init/autoconfig.js`), and its tests(`eslint/tests/lib/init/autoconfig.js`).
+- Remove dependency `progress`, as it's no longer used in production.
+
+### 2. Move `eslint --init` related files to a separate repo
+
+- Move the `eslint/lib/init/*` to the new repo's `lib/*` directory
+- Add the following `dependencies` in `package.json`
+
+  - `enquirer`
+  - `progress`
+  - `semver`
+  - `lodash`
```

Comment:
`lodash` has been replaced with other libraries, but I think it was used only in autoconfig so we can probably just remove this line.

Reply:
removed in bdc12aa062750d837e5a3bbbf2f6e5e3a98da388